addons/binding/org.openhab.binding.nibeuplink/ESH-INF/thing/base-channel-types.xml

```diff
@@ -0,0 +1,209 @@
+		<state pattern="%.1f °C" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-43081">
+		<item-type>Number:Time</item-type>
+		<label>Tot. op.time add.</label>
+		<description>Total electric additive operation time</description>
+		<state pattern="%.1f h" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-43084">
+		<item-type>Number:Power</item-type>
+		<label>Int. el.add. Power</label>
+		<description>Current power from the internal electrical addition</description>
+		<state pattern="%.2f kW" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-43161">
+		<item-type>Number</item-type>
+		<label>External adjustment activated via input S1</label>
+		<description></description>
+		<state pattern="%d" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-44298">
+		<item-type>Number:Energy</item-type>
+		<label>Heat Meter - HW Cpr and Add EP14</label>
+		<description>Accumulated energy production as calculated by the heat meter</description>
+		<state pattern="%.1f kWh" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-44300">
+		<item-type>Number:Energy</item-type>
+		<label>Heat Meter - Heat Cpr and Add EP14</label>
+		<description>Accumulated energy production as calculated by the heat meter</description>
+		<state pattern="%.1f kWh" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-44304">
+		<item-type>Number:Energy</item-type>
+		<label>Heat Meter - Pool Cpr EP14</label>
+		<description>Accumulated energy production as calculated by the heat meter</description>
+		<state pattern="%.1f kWh" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-44306">
+		<item-type>Number:Energy</item-type>
+		<label>Heat Meter - HW Cpr EP14</label>
+		<description>Accumulated energy production as calculated by the heat meter</description>
+		<state pattern="%.1f kWh" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-44308">
+		<item-type>Number:Energy</item-type>
+		<label>Heat Meter - Heat Cpr EP14</label>
+		<description>Accumulated energy production as calculated by the heat meter</description>
+		<state pattern="%.1f kWh" readOnly="true">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-47041">
+		<item-type>Number</item-type>
+		<label>Hot water mode</label>
+		<description> 0=Economy 1=Normal 2=Luxury</description>
+		<state pattern="%d" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-47212">
+		<item-type>Number:Power</item-type>
+		<label>Max int add. power</label>
+		<description></description>
+		<state min="0" max="4500" step="1" pattern="%.2f kW" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="base-type-48043">
+		<item-type>Number</item-type>
+		<label>Holiday</label>
+		<description>0=inactive, 10=active</description>
```

Comment:
It could be a switch. But a switch is 0/1 while this one is 10 / 0 So it does not really apply here. At least it would be necessary to have an additional mapping rule.

Reply:
Yes, technically the device expects  those number values. But from an end user point of view this is just switching holiday mode on/off. This way the user has to enter an integer into a text field and only 0/10 are allowed. The mapping rule should be just inside the binding code.
The binding is supposed to encapsulate all the technical details and present an end user interface on the channel/item side.

Reply:
This is already done. It is now a switch item.